package.json

```diff
@@ -28,31 +35,53 @@
     "fetch"
   ],
   "scripts": {
-    "prepublish": "npm run build && npm run types:check && npm run format:check",
-    "build": "tsup",
-    "test": "./setup-test-data.sh && tsc --project test/tsconfig.json",
+    "prepare": "husky",
+    "prepublishOnly": "npm run build && npm run exports:check && npm run types:check && npm run format:check && npm test",
+    "build": "del ./dist && tsc --project tsconfig.build.json",
+    "test": "vitest run --typecheck",
+    "dev": "vitest --typecheck",
+    "lint": "eslint './src/**/*'",
     "types:check": "tsc --noEmit",
     "format": "prettier --write .",
-    "format:check": "prettier --check ."
+    "format:check": "prettier --check .",
+    "exports:check": "attw --pack . --ignore-rules=cjs-resolves-to-esm"
   },
   "dependencies": {
-    "openapi-typescript-helpers": "0.0.8",
-    "type-fest": "4.23.0"
+    "openapi-typescript-helpers": "0.0.13",
+    "type-fest": "4.26.1"
   },
   "devDependencies": {
-    "@types/node": "22.0.0",
-    "expect-type": "0.19.0",
-    "openapi-fetch": "0.10.4",
-    "openapi-typescript": "7.1.2",
+    "@arethetypeswrong/cli": "0.16.2",
+    "@eslint/js": "9.10.0",
+    "@types/eslint__js": "8.42.3",
+    "@types/eslint-config-prettier": "6.11.3",
+    "@types/lodash": "4.17.7",
+    "@types/react": "18.3.5",
+    "@vitest/eslint-plugin": "1.1.0",
+    "del-cli": "5.1.0",
+    "eslint": "9.10.0",
+    "eslint-config-prettier": "9.1.0",
+    "husky": "9.1.5",
+    "lint-staged": "15.2.10",
+    "lodash": "4.17.21",
+    "openapi-fetch": "0.12.0",
+    "openapi-typescript": "7.4.0",
     "prettier": "3.3.3",
+    "react": "18.3.1",
     "swr": "2.2.5",
-    "tsup": "8.2.3",
-    "typescript": "5.5.4"
+    "typescript": "5.5.4",
+    "typescript-eslint": "8.5.0",
+    "vitest": "2.0.5"
   },
   "peerDependencies": {
-    "openapi-fetch": "0.10",
+    "openapi-fetch": "0.12",
     "openapi-typescript": "7",
-    "swr": "2",
-    "typescript": "5"
+    "react": "18 || 19",
+    "swr": "2"
+  },
+  "peerDependenciesMeta": {
+    "openapi-typescript": {
```

Comment:
How would types be generated without this? Is it really optional? I guess it could be used through CLI and not installed?

Reply:
That is right. This library isn't in charge of type generation, which is why I add it as an optional peer. I might even be able to remove it altogether, but I'd need to double check.

Reply:
After digging deeper, I think leaving in `openapi-typescript` as an optional peer dependency makes sense. Even though `swr-openapi` does not require it, the types it uses are expected to have been generated by `openapi-typescript@7`.